device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/fileupload/FileUploadTask.java

```diff
@@ -199,9 +208,31 @@ public IotHubStatusCode sendNotification(FileUploadCompletionNotification fileUp
         ResponseMessage responseMessage = httpsTransportManager.sendFileUploadNotification(message);
         httpsTransportManager.close();
 
+        validateServiceStatusCode(responseMessage, "Failed to complete the file upload notification");
+
         return responseMessage.getStatus();
     }
 
+    private void validateServiceStatusCode(ResponseMessage responseMessage, String errorMessage) throws IOException
+    {
+        String responseMessagePayload = null;
+        if (responseMessage.getBytes() != null && responseMessage.getBytes().length > 0)
+        {
+            responseMessagePayload = new String(responseMessage.getBytes(), DEFAULT_IOTHUB_MESSAGE_CHARSET);
+        }
+
+        IotHubServiceException serviceException =
+            IotHubStatusCode.getConnectionStatusException(
```

Comment:
What happens when payload is null?

Reply:
Then the IotHubServiceException will not have an error message property. It will just be the exception type that contains useful information for the user